### DISCUSSION TITLE
Reference UTF characters in C++ by Tile Index after upgrading

### DISCUSSION THREAD

**Discussion by kevrenac** [libtcod/libtcod#138] [Q&A] [answered]
I am working on upgrading `libtcod` from an older version to the latest, following the guide [here](https://libtcod.readthedocs.io/en/latest/upgrading.html). After upgrading to use `tcod::load_tilesheet` over the old method, I can no longer display any non-ASCII characters on my tilesheet (which is CP437). I am just getting a blank spot where the character should be. When using the old libtcod version, I simply referenced the [tile index](https://python-tcod.readthedocs.io/en/latest/tcod/charmap-reference.html), but now referencing any non-ASCII character (such as the smiley faces, indices 1 and 2 in my tileset) no longer works. I have also updated to `tcod::print()` (updating from `putChar`).

I fully suspect there is something I am doing wrong, or need to update to the latest libtcod version to get this to work... I am just not sure what. Any help would be appreciated.

**Comment by HexDecimal** [comment, marked as answer]
`tcod::load_tilesheet` takes `tcod::CHARMAP_CP437` for a CP437 font.  This variable is actually a `std::array` that you can index with CP437 EASCII codepoints to get the Unicode codepoints.  Keep in mind that from now on you should exclusively use the Unicode codepoints to reference these characters, such as `0x263A` for the White Smiling Face.

`tcod::CHARMAP_CP437.at(tile_index)` returns the Unicode for that tile index.  If your tilesheet is filled then this will only result in blanks for space and null as expected.  For example `tcod::CHARMAP_CP437.at(1)` returns the Unicode for White Smiling Face.

**Reply by kevrenac** [reply]
Thank you so much! That clears it up for me.
